src/main/java/com/mraof/minestuck/inventory/musicplayer/MusicPlayerContainer.java

```diff
@@ -0,0 +1,67 @@
+package com.mraof.minestuck.inventory.musicplayer;
+
+import com.mraof.minestuck.inventory.ContainerHelper;
+import com.mraof.minestuck.inventory.MSContainerTypes;
+import com.mraof.minestuck.inventory.musicplayer.ItemStackHandlerMusicPlayer;
+import com.mraof.minestuck.util.MSTags;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.SlotItemHandler;
+
+import javax.annotation.Nonnull;
+
+public class MusicPlayerContainer extends AbstractContainerMenu
+{
+	private static final int CASSETTE_X = 80;
+	private static final int CASSETTE_Y = 36;
+	
+	public MusicPlayerContainer(int windowId, Inventory playerInventory, IItemHandler itemStackHandler)
+	{
+		super(MSContainerTypes.MUSIC_PLAYER, windowId);
+		addSlot(new SlotItemHandler(itemStackHandler, 0, CASSETTE_X, CASSETTE_Y));
+		ContainerHelper.addPlayerInventorySlots(this::addSlot, 8, 84, playerInventory);
+	}
+	
+	public MusicPlayerContainer(int windowId, Inventory playerInventory)
+	{
+		this(windowId, playerInventory, new ItemStackHandlerMusicPlayer(1));
+	}
+	
+	@Override
+	public boolean stillValid(Player playerIn)
+	{
+		return true;
```

Comment:
I realize that you may need to hold on to the lazy optional to some degree in order to check if the capability still exists, instead of potentially holding on to an item handler that may become invalidated without the container knowing it. Probably worthwhile to check on some example elsewhere in the modding community on what might be needed.